src/gyroaverage/gyroaverage_operator.hpp

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include <ddc/ddc.hpp>
+
+namespace detail {
+
+/**
+ * @brief Operator to compute the gyroaverage of a field in (r, theta) coordinates.
+ *
+ * This class performs the gyroaveraging operation on a batched field defined on a polar grid (r, theta).
+ * The gyroaverage is computed by integrating the field over a set of points along a circle (the Larmor orbit)
+ * centred at each grid point, with radius given by the local Larmor radius field (rho_L).
+ *
+ * The class uses 2D B-spline interpolation to evaluate the field at off-grid points along the orbit.
+ * The operation is performed in parallel over the (r, theta) grid.
+ *
+ * @tparam ExecutionSpace      The Kokkos execution space.
+ * @tparam GridR               The radial grid type.
+ * @tparam GridTheta           The poloidal angle grid type.
+ * @tparam GridBatch           The batch grid type (for batched operations).
+ * @tparam BSplinesR           The B-spline basis for the radial direction.
+ * @tparam BSplinesTheta       The B-spline basis for the theta direction.
+ */
+template <
+        class ExecutionSpace,
+        class GridR,
+        class GridTheta,
+        class GridBatch,
```

Comment:
There is a single batch dimension right ?

Reply:
Could you not pass an ND IndexRange instead of a Grid in that case?

Reply:
To minimize the size of the PR, I would like to do that in another PR. The treatment of batch dimension is less important compared to the averaging operation in r and theta.

Reply:
Passing an index range (usually one equivalent to your `IdxRangeRThetaBatch`) is the standard way of doing this in Gyselalib++. I don't think it would add much code to your PR (just a couple of lines to get `IdxRangeBatch`).

If you don't want to handle batching in this PR then I would prefer that we remove `GridBatch`

Reply:
```cpp
using IdxRangeBatch = ddc::remove_dims_of_t<IdxRangeRThetaBatch, GridR, GridTheta>;
```

Reply:
How to access `GridBatch` tag from `IdxRangeBatch`?

Reply:
> How to access GridBatch tag from IdxRangeBatch?

You can't. The aim is to hide this so that IdxRangeBatch can be `IdxRange<>`, `IdxRange<GridBatch>`, `IdxRange<GridBatch1, GridBatch2>`, ...